### PR TITLE
SnsTemplate provides functionality to validate topic existence (#1137)

### PR DESCRIPTION
**Pull request by hardikSinghBehl**
## :loudspeaker: Type of change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description

Implemented the method `boolean topicExists(String topicArn)` in `SnsTemplate` to validate the existence of an SNS topic via an ARN.

## :bulb: Motivation and Context

Implementation for issue #1137

## :green_heart: How did you test it?
The functionality has been tested by writing integration tests for the new method. 

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] I updated reference documentation to reflect the change
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

**Comment by MatejNedic** [comment]
Hey @hardikSinghBehl thanks on PR! Will take a deeper look tomorrow and come back to you

File: .gitignore [hunk 1]
```diff
@@ -20,6 +20,7 @@ mail.properties
 .project
 .classpath
 .settings/
+.DS_Store
```
Comment (MatejNedic) [inline .gitignore:23]: Please remove this

**Review by MatejNedic — CHANGES_REQUESTED** [review]
All good, just remove that 1 thing from gitignore

**Comment by hardikSinghBehl** [comment]
done, @MatejNedic

**Review by MatejNedic — APPROVED** [review]

**Comment by maciejwalkowiak** [comment]
Thanks @hardikSinghBehl!

**Pull request merged**